src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1577,9 +1577,7 @@ void enableSSL(String host,
         String tmfDefaultAlgorithm = null;  // Default algorithm (typically X.509) used by the TrustManagerFactory
         SSLHandhsakeState handshakeState = SSLHandhsakeState.SSL_HANDHSAKE_NOT_STARTED;
 
-        boolean isFips = false;
         String trustStoreType = null;
-        String fipsProvider = null;
```

Comment:
Hi @nsidhaye ,
Thank you so much for detailed comments. We appreciate the time and effort. 

As I mentioned above, if `fipsProvider` is not specified,`KeyStore.getInstance()` traverses the list of registered security Providers - `Security.getProviders()` and returns the matching `KeyStore` object. 

Example, in case of option 3, user would make sure `BouncyCastleFipsProvider` is installed before creating a connection.

```
Security.addProvider(new org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider());
Security.addProvider(new com.sun.net.ssl.internal.ssl.Provider ("BCFIPS"));
ds.setTrustStoreType("BCFKS");
connection = (SQLServerConnection) ds.getConnection();
```
When the drivers calls `KeyStore.getInstance(trustStoreType)` **before TLS Handshake**, `BCFIPS` will already be in the list of installed providers.